### ISSUE TITLE
DBW: suggest CSS containment

### ISSUE THREAD

**Issue by ebidel** [GoogleChrome/lighthouse#719]
Discoverability:
App is doing excessive painting, layout, and style recalcs and invalidating styles on large regions.

**Comment by wardpeet** [comment]
YES!

**Comment by Seirdy** [comment]
Agreed; this could also include improvements to the "avoid excessive DOM size" audit. That audit should take into account the DOM nodes rendered on load and/or the max DOM nodes rendered at once, not the total DOM nodes on the page.

**Comment by Seirdy** [comment]
Specifically, LH should suggest using `content-visibility: auto` rather than `content-visibility: hidden`. `auto` is accessible to users who search for text in a page (e.g. when using <kbd>Ctrl</kbd>+<kbd>F</kbd>) and assistive technologies.

That being said, *it should only recommend this for sections at the end of the page* to avoid introducing layout shifts. Using containment in the middle of the page would require computing the content's intrinsic size, which can be extremely challenging when sites delegate formatting to the user agent.

Notable exceptions: content like complex SVGs with known dimensions can easily be contained without introducing layout shifts.
